src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala

```diff
@@ -404,4 +405,352 @@ class WorkspaceServiceSpec extends FlatSpec with ScalatestRouteTest with Matcher
+    assertResult(Vector(MethodConfigurationShort("testConfig1","Sample",MethodRepoMethod("ns-config","meth1",1),"ns"))) {
+      runAndWait(methodConfigurationQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission)))
+    }
+
+    //Check if access levels on workspace exist
+    assertResult(Vector(("owner-access", WorkspaceAccessLevels.Owner), ("reader-access",WorkspaceAccessLevels.Read), ("writer-access",WorkspaceAccessLevels.Write))) {
+      runAndWait(workspaceQuery.listEmailsAndAccessLevel(SlickWorkspaceContext(testData.workspaceFailedSubmission)))
+    }
+
+    //Check if submissions on workspace exist
+    assertResult(List(testData.submissionFailed)) {
+      runAndWait(submissionQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission)))
+    }
+
+    //Check if entities on workspace exist
+    assertResult(Vector(testData.indiv1, testData.indiv2)) {
+      runAndWait(entityQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission), "Individual"))
+    }
+    assertResult(Vector(testData.sample1, testData.sample2, testData.sample3, testData.sample4, testData.sample5, testData.sample6, testData.sample7, testData.sample8)) {
+      runAndWait(entityQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission), "Sample"))
+    }
+    assertResult(Vector(testData.aliquot1, testData.aliquot2)) {
+      runAndWait(entityQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission), "Aliquot"))
+    }
+    assertResult(Vector(testData.pair1, testData.pair2)) {
+      runAndWait(entityQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission), "Pair"))
+    }
+    assertResult(Vector(testData.ps1)) {
+      runAndWait(entityQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission), "PairSet"))
+    }
+    assertResult(Vector(testData.sset1, testData.sset2, testData.sset3, testData.sset4, testData.sset_empty)) {
+      runAndWait(entityQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission), "SampleSet"))
+    }
+
+    //delete the workspace
+    Await.result(services.workspaceService.deleteWorkspace(testData.wsName5), Duration.Inf)
+
+    //check that the workspace has been deleted
+    assertResult(None) {
+      runAndWait(workspaceQuery.findByName(testData.wsName5))
+    }
+
+    //check if method configs have been deleted
+    assertResult(Vector()) {
+      runAndWait(methodConfigurationQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission)))
+    }
+
+    //Check if access levels have been deleted
+    assertResult(Vector()) {
+      runAndWait(workspaceQuery.listEmailsAndAccessLevel(SlickWorkspaceContext(testData.workspaceFailedSubmission)))
+    }
+
+    //Check if submissions on workspace have been deleted
+    assertResult(Vector()) {
+      runAndWait(submissionQuery.list(SlickWorkspaceContext(testData.workspaceFailedSubmission)))
+    }
+
+    //Check if entities on workspace have been deleted
```

Comment:
I think you can just assert `runAndWait(entityQuery.length.result))` == 0, rather than checking them all manually